resources/aws_ec2.go

```diff
@@ -156,6 +156,11 @@ type AwsEc2Res struct {
 	// for documantation and examples.
 	UserData string `yaml:"userdata"`
 
+	// PublicIP is reserved for the instance's public IPv4 address and is
+	// updated by CheckApply() whenever it is run. This is not a parameter
```

Comment:
Maybe instead of "This is not a parameter that should be defined by the user." how about "This is a read-only parameter." or similar?

Can we call this PublicIPv4 instead ?
I'm assuming there's always a PrivateIPv4 as well that we could get easily? (If so want to add that as well?)